test/jdk/tools/jpackage/macosx/SigningRuntimeImagePackageTest.java

```diff
@@ -0,0 +1,263 @@
+            if (dmgImage.endsWith(cmd.name() + ".jdk")) {
+                SigningBase.verifyCodesign(launcherPath, isRuntimeImageSigned,
+                        JDKBundleCertIndex);
+                SigningBase.verifyCodesign(dmgImage, isRuntimeImageSigned,
+                        JDKBundleCertIndex);
+                if (isRuntimeImageSigned) {
+                    SigningBase.verifySpctl(dmgImage, "exec", JDKBundleCertIndex);
+                }
+            }
+        });
+    }
+
+    private static boolean isPKGSigned(JPackageCommand cmd) {
+        return cmd.hasArgument("--mac-signing-key-user-name") ||
+               cmd.hasArgument("--mac-installer-sign-identity");
+    }
+
+    private static int getCertIndex(JPackageCommand cmd) {
+        if (cmd.hasArgument("--mac-signing-key-user-name")) {
+            String devName = cmd.getArgumentValue("--mac-signing-key-user-name");
+            return SigningBase.getDevNameIndex(devName);
+        } else {
+            return SigningBase.CertIndex.INVALID_INDEX.value();
+        }
+    }
+
+    private static Path getRuntimeImagePath(boolean useJDKBundle,
+                                            boolean isRuntimeImageSigned,
+                                            int JDKBundleCertIndex) throws IOException {
+        final Path runtimeBundleDir =
+                TKit.createTempDirectory("runtimebundle");
+        final Path runtimeImageImage =
+                runtimeBundleDir.resolve("image");
+
+        new Executor()
+            .setToolProvider(JavaTool.JLINK)
+            .dumpOutput()
+            .addArguments(
+                "--output", runtimeImageImage.toString(),
+                "--add-modules", "java.desktop",
+                "--strip-debug",
+                "--no-header-files",
+                "--no-man-pages")
+            .execute();
+
+        if (useJDKBundle) {
+            // We will use jpackage to create JDK bundle from image signed or
+            // unsigned.
+
+            final Path runtimeBundleDMG =
+                runtimeBundleDir.resolve("dmg");
+            final Path runtimeBundleBundle =
+                runtimeBundleDir.resolve("bundle");
+
+            Executor ex = new Executor();
+            ex.setToolProvider(JavaTool.JPACKAGE)
+            .dumpOutput()
+            .addArguments(
+                "--type", "dmg",
+                "--name", "foo",
+                "--runtime-image", runtimeImageImage.toAbsolutePath().toString(),
+                "--dest", runtimeBundleDMG.toAbsolutePath().toString());
+
+            if (isRuntimeImageSigned) {
+                ex.addArguments(
+                    "--mac-sign",
+                    "--mac-signing-keychain", SigningBase.getKeyChain(),
+                    "--mac-signing-key-user-name", SigningBase.getDevName(JDKBundleCertIndex));
+            }
+
+            ex.execute();
+
+            JPackageCommand dummyCMD = new JPackageCommand();
```

Comment:
Why do you need a `dummyCMD`? You can configure a normal JPackageCommand command:
```
var cmd = new JPackageCommand().useToolProvider(true).dumpOutput(true).addArguments(...);
```